prow/flagutil/job_config.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+
+	"k8s.io/test-infra/prow/config"
+)
+
+// JobConfigOptions holds options for interacting with job configs.
+type JobConfigOptions struct {
+	JobConfigPath string
+}
+
+// AddFlags injects job config options into the given FlagSet.
+func (o *JobConfigOptions) AddFlags(fs *flag.FlagSet) {
+	fs.StringVar(&o.JobConfigPath, "job-config-path", "", "Path to prow job configs.")
```

Comment:
Do we have a use-case for only loading job config and not core config? `hook`?

Also, since many will need both -- can we implement something (maybe using struct embedding) that gives downstream callers a simple way to request both flags without having downstream code to funnel the configPath from one place to another? Hopefully we can make the actual paths opaque to the callers (and unexported)